nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -210,6 +210,20 @@ Costmap2DROS::on_configure(const rclcpp_lifecycle::State & /*state*/)
     layered_costmap_->getCostmap(), global_frame_,
     "costmap", always_send_full_costmap_);
 
+  auto layers = layered_costmap_->getPlugins();
```

Comment:
Costmap Filters could also contain costmap layers (see https://github.com/ros-planning/navigation2/issues/3102#issuecomment-1225494691 for more details). However, since by design, these costmap layers are intended for costmap filters, so we really might not be interested in, at least in this PR.